SKA-ScienceDataProcessor/integration-prototype
# log error if target state is not allowed

## Description:
The master controller logs an error and leaves the current state unchanged if the target state is set to a state that can't be transitioned to. Fixes issue #91 


- `is_target_state_allowed` method added to StateObject class
- in `_process_event` in `master_controller` an error is logged if the target state is not allowed in the
current state and the state left unchanged.

## Testing instructions:
Follow instructions in `demos/01_sdp_states` and set the target state to some not-allowed state. Observe that the master controller logs an error and the current state does not change.

## Types of changes


## Checklist:
